lib/job-iteration/iteration.rb

```diff
@@ -251,5 +287,21 @@ def valid_cursor_parameter?(parameters)
       end
       false
     end
+
+    def serializable?(object)
+      # Subclasses must be excluded, hence not using is_a? or ===.
+      if object.instance_of?(Array)
+        object.all? { |element| serializable?(element) }
+      elsif object.instance_of?(Hash)
+        object.all? { |key, value| serializable?(key) && serializable?(value) }
+      else
+        object.instance_of?(String) ||
+          object.instance_of?(Integer) || object.instance_of?(Float) ||
+          object.equal?(nil) || object.equal?(true) || object.equal?(false)
```

Comment:
We could probably dry the code a little here by doing:
```ruby
klasses = [String, Integer, Float, NilClass, TrueClass, FalseClass]
klasses.any? { |klass| object.instance_of?(klass) }
```

We might be even be able to extract `klasses` as a constant

Reply:
It's funny, I actually went through having both those implementations but got caught up in trying to optimize for speed because this will run before `each_iteration`. 😅 

I figured we could unroll the loop, and use `.equal?` to quickly compare by object ID for singletons, instead of looking at their class. But benchmarking it shows a negligible difference, so I'll go with the constant array approach for readability. 👍